lib/internal/Magento/Framework/Validator/Factory.php

```diff
@@ -31,40 +42,25 @@ class Factory
     private $isDefaultTranslatorInitialized = false;
 
     /**
-     * @var \Magento\Framework\Module\Dir\Reader
+     * @var Reader
      */
     private $moduleReader;
 
-    /**
-     * @var FrontendInterface
-     */
-    private $cache;
-
-    /**
-     * @var \Magento\Framework\Serialize\SerializerInterface
-     */
-    private $serializer;
-
-    /**
-     * @var \Magento\Framework\Config\FileIteratorFactory
-     */
-    private $fileIteratorFactory;
-
     /**
      * Initialize dependencies
      *
-     * @param \Magento\Framework\ObjectManagerInterface $objectManager
-     * @param \Magento\Framework\Module\Dir\Reader $moduleReader
-     * @param FrontendInterface $cache
+     * @param ObjectManagerInterface $objectManager
+     * @param Reader $moduleReader
+     * @param FrontendInterface $cache @deprecated
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
-        \Magento\Framework\ObjectManagerInterface $objectManager,
-        \Magento\Framework\Module\Dir\Reader $moduleReader,
+        ObjectManagerInterface $objectManager,
```

Comment:
You should mantain the Backward compatibility adding a null value or creating by Object Manager in construct

Reply:
Hi @osrecio,

thanks for your feedback. The constructor arguments stay exactly the same as they were (I just imported the class names). The only actual change in the constructor is the `@deprecation` notice and the removed property `$this->cache`.

As it is a private property, I think it can be removed safely.

Would you please clarify what you mean?

Reply:
Ohh yes, was my fault, you are rigth